BuildMasterAutomation/Functions/Remove-BMServerRole.ps1

```diff
@@ -0,0 +1,44 @@
+
+function Remove-BMServerRole
+{
+    <#
+    .SYNOPSIS
+    Removes a server role from BuildMaster.
+
+    .DESCRIPTION
+    The `Remove-BMServerRole` removes a server role from BuildMaster. Pass the name of the role to remove to the `Name` parameter. If the server role doesn't exist, nothing happens.
+
+    Pass the session to the BuildMaster instance where you want to delete the role to the `Session` parameter. Use `New-BMSession` to create a session object.
+
+    This function uses BuildMaster's infrastructure management API.
+
+    .EXAMPLE
+    Remove-BMServerRole -Session $session -Name 'Server Role'
+
+    Demonstrates how to delete a server role.
+
+    .EXAMPLE
+    Get-BMServerRole -Session $session -Name 'My Role' | Remove-BMServerRole -Session $session
+
+    Demonstrates that you can pipe the objects returned by `Get-BMServerRole` into `Remove-BMServerRole` to remove those roles.
+    #>
+    [CmdletBinding(SupportsShouldProcess)]
+    param(
+        [Parameter(Mandatory)]
+        # The instance of BuildMaster to connect to.

```

Comment:
Nit: Inconsistent parameter doc language. The `$Session` parameter doc usually says: `# An object representing the instance of BuildMaster to connect to. Use  New-BMSession to create session objects.`